Sources/Backup/OABackupHelper.mm

```diff
@@ -569,6 +581,157 @@ - (BOOL) prepareBackup
     return YES;
 }
 
+- (NSString *) uploadFile:(NSString *)fileName
+                     type:(NSString *)type
+                     data:(NSData *)data
+               uploadTime:(NSTimeInterval)uploadTime
+                 listener:(id<OAOnUploadFileListener>)listener
+{
+    [self checkRegistered];
+    
+    NSMutableDictionary<NSString *, NSString *> *params = [NSMutableDictionary dictionary];
+    params[@"deviceid"] = [self getDeviceId];
+    params[@"accessToken"] = [self getAccessToken];
+    params[@"name"] = fileName;
+    params[@"type"] = type;
+    params[@"clienttime"] = [NSString stringWithFormat:@"%.0f", uploadTime];
+    
+    NSMutableDictionary<NSString *, NSString *> *headers = [NSMutableDictionary dictionary];
+    headers[@"Accept-Encoding"] = @"deflate, gzip";
+    
+//    OperationLog operationLog = new OperationLog("uploadFile", DEBUG);
+//    operationLog.startOperation(type + " " + fileName);
+    __block NSString *error = nil;
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    [OANetworkUtilities uploadFile:UPLOAD_FILE_URL fileName:fileName params:params headers:headers data:data gzip:YES onComplete:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable err) {
+        if (((NSHTTPURLResponse *)response).statusCode != 200)
+        {
+            error = data ? [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding] : nil;
+        }
+        dispatch_semaphore_signal(semaphore);
```

Comment:
Would be better to move semaphores inside OANetworkUtilities. For example make two methods like `uploadFile` and `uploadFileAsync`, or introduce parameter `wait`.